client/verta/verta/client.py

```diff
@@ -1836,7 +1873,15 @@ class ExperimentRun(_ModelDBEntity):
         Name of this Experiment Run.
 
     """
-    def __init__(self, conn, conf,
+    def __init__(self, conn, conf, id_):
+        super(ExperimentRun, self).__init__(conn, conf, _ExperimentRunService, "experiment-run", id_)
+
+        self._conn = conn
+
+        self.id = id_
+
+    @staticmethod  # TODO: if PR approved, move this fn down with other static/class methods
+    def _get_or_create(conn, conf,
```

Comment:
Here's where the logic is moved out of `ExperimentRun.__init__()` and into `_get_or_create()`